translator/translate/logs/logs_collected/windows_events/collect_list/collectlist_test.go

```diff
@@ -67,3 +68,143 @@ func TestApplyRule(t *testing.T) {
 		panic(error)
 	}
 }
+
+func TestDuplicateRetention(t *testing.T) {
+	c := new(CollectList)
+	var rawJsonString = `
+{
+    "collect_list": [
+      {
+        "event_name": "System",
+        "event_levels": [
+          "INFORMATION",
+          "CRITICAL"
+        ],
+        "log_group_name": "System",
+		"retention_in_days": 3
+      },
+      {
+        "event_name": "Application",
+        "event_levels": [
+          "INFORMATION",
+          "VERBOSE",
+          "ERROR"
+        ],
+        "event_format": "xml",
+        "log_group_name": "System",
+		"retention_in_days": 3
+      },
+      {
+        "event_name": "Application",
+        "event_levels": [
+          "INFORMATION",
+          "VERBOSE",
+          "ERROR"
+        ],
+        "event_format": "xml",
+        "log_group_name": "System",
+		"retention_in_days": 3
+      }
+    ]
+}
+`
+	var input interface{}
+
+	var expected = []interface{}{
+		map[string]interface{}{
+			"event_name":        "System",
+			"event_levels":      []interface{}{"4", "0", "1"},
+			"log_group_name":    "System",
+			"batch_read_size":   BatchReadSizeValue,
+			"retention_in_days": 3,
+		},
+		map[string]interface{}{
+			"event_name":        "Application",
+			"event_levels":      []interface{}{"4", "0", "5", "2"},
+			"event_format":      "xml",
+			"log_group_name":    "System",
+			"batch_read_size":   BatchReadSizeValue,
+			"retention_in_days": -1,
+		},
+		map[string]interface{}{
+			"event_name":        "Application",
+			"event_levels":      []interface{}{"4", "0", "5", "2"},
+			"event_format":      "xml",
+			"log_group_name":    "System",
+			"batch_read_size":   BatchReadSizeValue,
+			"retention_in_days": -1,
+		},
+	}
+
+	var actual interface{}
+
+	error := json.Unmarshal([]byte(rawJsonString), &input)
+	if error == nil {
+		_, actual = c.ApplyRule(input)
+		assert.Equal(t, 0, len(translator.ErrorMessages))
+		assert.Equal(t, expected, actual)
+	} else {
+		panic(error)
```

Comment:
Why use `panic()` in test code?
Why not just fail an assert?

Reply:
There's a lot of test code like this. I was going to go through and clean up the test code so it doesn't panic. Personally I wouldn't block on this because I'm going to be cleaning up afterwards.

Reply:
Yeah it was copy paste pasta but i'll change it in the new test cases and we can clean up the the previous code later like Andrew said.